src/ContextFreeTasks.Shared/ContextFreeTask.cs

```diff
@@ -1,13 +1,15 @@
 ﻿using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
+using static System.Threading.Tasks.Task;
 
 namespace ContextFreeTasks
 {
     [AsyncMethodBuilder(typeof(AsyncContextFreeTaskMethodBuilder))]
     public struct ContextFreeTask
     {
-        public Task Task { get; }
-        internal ContextFreeTask(Task t) => Task = t;
+        private readonly Task _task;
+        public Task Task => _task ?? FromResult(default(object));
+        internal ContextFreeTask(Task t) => _task = t;
```

Comment:
Yes, I concerned about `default(ContextFreeTask)`. if provides struct as public, we must consider the default.
for example, `ValueTask` created with default keyword, it is same as `FromResult(default(T))`.
the constructor of `ContextFreeTask` is hid to internal. so ONLY `default(ContextFreeTask)` makes `_task` be `null`. it's very rare case.